pkg/web/http.go

```diff
@@ -21,12 +22,18 @@ type HTTPServer struct {
 func NewHTTPServer(ctx context.Context) (*HTTPServer, error) {
 	cfg := config.FromContext(ctx)
 	logger := log.FromContext(ctx)
+
+	CORSHeaders := handlers.AllowedHeaders([]string{"Content-Type", "X-Requested-With", "User-Agent", "Authorization"})
+	CORSOrigins := handlers.AllowedOrigins([]string{"*"})
+	CORSMethods := handlers.AllowedMethods([]string{"GET", "HEAD", "POST", "PUT", "OPTIONS"})
+	handlerWithCORS := handlers.CORS(CORSHeaders, CORSOrigins, CORSMethods)(NewRouter(ctx))
```

Comment:
You probably want to put these lines here https://github.com/charmbracelet/soft-serve/blob/main/pkg/web/server.go#L27